ui/src/App.tsx

```diff
@@ -22,7 +22,7 @@ const RedirectToFirstPaper = () => {
     useEffect(() => {
         getAllocatedPaperStatus().then((allocation) => {
             const first = allocation.papers[0];
-            setSha(first.sha);
+            if (first) setSha(first.sha);
```

Comment:
This is a functional change which seems unrelated to the other changes in this PR. Is it intentional?

Reply:
So, this is throwing an error when we trying to start app first time and didn't put some pdf's into skiff folder. Should I create separate PR with this fix and implementation of case described above (show some page that says "there is no pdf's")

Reply:
> Should I create separate PR with this fix and implementation of case described above (show some page that says "there is no pdf's" or something like this)

Yea, that'd be great (and seems like a good addition). Thank you for taking this on!

Reply:
I think you can revert this change, now that your other PR has merged. Thx!